napari/layers/utils/link_layers.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+
+#: Record of already linked layers... to avoid duplicating callbacks
+#  in the form of (id(layer1), id(layer2), attribute_name)
+_LINKED: set[tuple[int, int, str]] = set()
+
+
+# attributes that make no sense to link, or error on changing
+# we may want to include colormap here, but in most cases, it works fine.
+PROHIBITED = {'thumbnail', 'status', 'name'}
+
+
+def _get_common_evented_attributes(
+    layers: Iterable['Layer'],
+    prohibited: set[str] = PROHIBITED,
+    with_private=False,
+) -> set[str]:
+    """Get the set of common, non-private evented attributes in ``layers``.
+
+    Not all layer events are attributes, and not all attributes have
+    corresponding events.  Here we get the set of valid, non-private attributes
+    that are both events and attributes for the provided layer set.
+
+    Parameters
+    ----------
+    layers : iterable
+        A set of layers to evaluate for attribute linking.
+    prohibited : set, optional
```

Comment:
Can we use `exclude` as a kwarg instead of `prohibited` - Pydantic uses exclude as part of its [serialization/ to dict API](https://pydantic-docs.helpmanual.io/usage/exporting_models/#modeldict) and I have a slight preference for it

Reply:
yep!